packages/matter.js/tsconfig.json

```diff
@@ -7,6 +7,7 @@
     "incremental": true, // Speed up subsequent builds
     "moduleResolution": "node16", // Required to correctly resolve Babel imports and to make sure all imports are js
     "module": "node16", // Required to make sure all imports are js
+    "target": "ES2018",
```

Comment:
I'm still not too happy with that because it might lead devs in wrong securoty about support, or? And then we generate for another version ... Where do we need these features?

Reply:
IIRC at one point ts-node was bombing on object spreads without this.  I'll try taking it out and see what happens.

Reply:
Followup on this, I was not able to get by entirely without but ES2017 works.  That's the minimum required for running top-level asyncs in the utility scripts

Reply:
kk, alternative could be an own tsconfig fpr just this directory tree where we need it which uises the base file and just overrides this one for these sub directories?

Reply:
That would be OK, would make command line invocation uglier but it's already ugly without npm because of ts-node.

WDYT about current target of ES6 vs ES8?  Is there an environment we target that doesn't support async/await?  Other than that I think support for ES6 is pretty much a lie...  AFAICT only other significant ES6->ES8 deltas are array.includes, string.padStart/padEnd and Object.entries/values/etc. all of which we use extensively and don't transpile or shim

Reply:
I'm resolving this here because moving the files out of matter.js made this irrelevant for this commit.  But the point stands, we should probably change our target to ES8 rather than ES6.